File: backend/src/routes/profile.routes.ts
import { Router } from 'express'
import { getMyProfile, updateMyProfile, getAllProfiles, updateUserRole } from '../controllers/profile.controller'
import { requireAuth, requireAdmin } from '../middlewares/auth.middleware'

const router = Router()

// Todas as rotas de perfil exigem login (requireAuth vem primeiro)
// O fluxo é: requisição → requireAuth → controller

// GET  /api/profile/me  → retorna o perfil do usuário logado
router.get('/me', requireAuth, getMyProfile)

// PUT  /api/profile/me  → atualiza nome, foto ou bio do usuário logado
router.put('/me', requireAuth, updateMyProfile)

// GET  /api/profile/all → [só admin] lista todos os usuários
router.get('/all', requireAuth, requireAdmin, getAllProfiles)

// PUT  /api/profile/:id/role → [só admin] muda o role de um usuário
router.put('/:id/role', requireAuth, requireAdmin, updateUserRole)

export default router

File: backend-go/internal/profile/profile_routes.go
package profile

import (
	"github.com/gin-gonic/gin"
	"github.com/jackc/pgx/v5/pgxpool"

	"beeplanerofc/backend-go/internal/middleware"
	"beeplanerofc/backend-go/internal/supabaseauth"
)

// RegisterRoutes registra as rotas de perfil no grupo /api/profile.
// Todas exigem login (requireAuth); /all e /:id/role exigem admin.
func RegisterRoutes(router *gin.RouterGroup, db *pgxpool.Pool, authClient *supabaseauth.Client) {
	ctrl := NewController(db)

	router.Use(middleware.RequireAuth(db, authClient))

	router.GET("/me", ctrl.GetMyProfile)
	router.PUT("/me", ctrl.UpdateMyProfile)
	router.GET("/all", middleware.RequireAdmin, ctrl.GetAllProfiles)
	router.PUT("/:id/role", middleware.RequireAdmin, ctrl.UpdateUserRole)
}
